src/tvheadend/entity/Event.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+/*
+ *      Copyright (C) 2005-2011 Team XBMC
+ *      http://www.xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, write to
+ *  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ *  http://www.gnu.org/copyleft/gpl.html
+ *
+ */
+
+#include "Entity.h"
+#include <map>
+#include <cstdint>
+#include <string>
+
+namespace tvheadend
+{
+  namespace entity
+  {
+    class Event;
+    typedef std::pair<uint32_t, Event> EventMapEntry;
+    typedef std::map<uint32_t, Event> Events;
+
+    /**
+     * Represents an event/programme
+     */
+    class Event : public Entity
+    {
+    public:
+      Event() :
+        Entity(),
```

Comment:
No need to call the base class default ctor directly. Will be done automatically. Here and elsewhere...